synapse/lib/nexus.py

```diff
@@ -154,19 +196,108 @@ def getOffsetEvent(self, offs):
                 yield item
 
     @contextlib.asynccontextmanager
-    async def getChangeDist(self, offs):
+    async def getChangeDist(self, offs: int) -> AsyncIterator[ChangeDist]:
 
-        async with await ChangeDist.anit(self.nexuslog, offs) as dist:
+        async with await ChangeDist.anit(self._nexuslog, offs) as dist:
 
             async def fini():
-                self.mirrors.remove(dist)
+                self._mirrors.remove(dist)
 
             dist.onfini(fini)
 
-            self.mirrors.append(dist)
+            self._mirrors.append(dist)
 
             yield dist
 
+    async def setLeader(self, url: Optional[str], iden: str) -> None:
+        '''
+        Args:
+            url:  if None, sets this nexsroot as leader, otherwise the telepath URL of the leader (must be a Cell)
+            iden: iden of the leader.  Should be the same as my containing cell's iden
+        '''
+        if url is not None and not self.dologging:
+            raise s_exc.BadConfValu(mesg='Mirroring incompatible without logchanges')
+
+        if self._ldrurl == url:
+            return
+
+        self._ldrurl = url
+
+        if self._looptask is not None:
+            self._looptask.cancel()
+            self._looptask = None
+            self._ldrready.clear()
+            if self._ldr is not None:
+                await self._ldr.fini()
+            self._ldr = None
+
+        if self._ldrurl is None:
+            return
+
+        self._looptask = self.schedCoro(self._followerLoop(iden))
+
+    async def _followerLoop(self, iden) -> None:
+
+        while not self.isfini:
+
+            try:
+
+                async with await s_telepath.openurl(self._ldrurl) as proxy:
+                    self._ldr = proxy
+                    self._ldrready.set()
+
+                    # if we really are a mirror/follower, we have the same iden.
+                    if iden != await proxy.getCellIden():
+                        logger.error('remote cortex has different iden!  Aborting mirror sync')
+                        await self.fini()
```

Comment:
so, once this happens, what all gets torn down?  does the cell exit?

Reply:
That's what it did before. Should it?

Reply:
Doesn't this tearndown the NexusRoot object, and not the host cell?

Reply:
lets discuss this behavior today :)

Reply:
(Discussed in person) Answer: yes